x-pack/plugins/session_view/public/components/session_view/index.tsx

```diff
@@ -124,8 +128,20 @@ export const SessionView = ({
     isFetching: isFetchingAlerts,
     hasNextPage: hasNextPageAlerts,
     error: alertsError,
+    refetch: refetchAlerts,
   } = useFetchSessionViewAlerts(sessionEntityId, investigatedAlertId);
 
+  useEffect(() => {
+    refetch().then((val) => {
+      setRefreshSession(val.isLoading);
+    });
+    refetchAlerts();
+  }, [refreshSession, refetch, refetchAlerts]);
+
+  const handleRefresh = useCallback(() => {
```

Comment:
won't need to add it on the callbacks too, `refetchAlerts` and `refetch` is already memoized by react-query